### PR TITLE
Sync sieve docs with problem-specifications

### PR DESCRIPTION
**Pull request by kytrinyx**
The sieve exercise has been overhauled as part of a project to make practice exercises more consistent and friendly.

For more context, please see the discussion in the forum, as well as the pull request that updated the exercise in the problem-specifications repository:

- https://forum.exercism.org/t/new-project-making-practice-exercises-more-consistent-and-human-across-exercism/3943
- https://github.com/exercism/problem-specifications/pull/2216

### CODE REVIEW

**Review by meatball133 — APPROVED** [review]
Awsome! ✨

**Comment by kytrinyx** [comment]
@meatball133 in general if you're happy with the changes, would you go ahead and merge the PR? I will eventually merge approved PRs but it will take at least a week before I get to it, so if you merge it, it will get deployed much more quickly.

**Comment by meatball133** [comment]
@kytrinyx I merge as fast as the ci are done. I got an email saying a new pr was uploaded so approved it and I had to go and couldn't wait until ci were done. But I often merge these prs within 2 hours, as you may have noticed with earlier prs.

**Pull request merged**
